docs/specification/draft/basic/authorization.mdx

```diff
@@ -302,6 +338,53 @@ Servers **MUST** return appropriate HTTP status codes for authorization errors:
+
+##### Runtime Insufficient Scope Errors
+
+When a client makes a request with an access token with insufficient scope during runtime operations, the server **SHOULD** respond with:
+
+- `HTTP 403 Forbidden` status code (per [RFC 6750 Section 3.1](https://datatracker.ietf.org/doc/html/rfc6750#section-3.1))
+- `WWW-Authenticate` header with the `Bearer` scheme and additional parameters:
+  - `error="insufficient_scope"` - indicating the specific type of authorization failure
+  - `scope="required_scope1 required_scope2"` - specifying the minimum scopes needed for the operation
+  - `resource_metadata` - the URI of the Protected Resource Metadata document (for consistency with 401 responses)
+  - `error_description` (optional) - human-readable description of the error
+
+Example insufficient scope response:
+
+```http
+HTTP/1.1 403 Forbidden
+WWW-Authenticate: Bearer error="insufficient_scope",
+                         scope="files:read files:write",
+                         resource_metadata="https://mcp.example.com/.well-known/oauth-protected-resource",
+                         error_description="Additional file write permission required"
+```
+
+##### Client Scope Upgrade Flow
+
+Clients will receive scope-related errors during initial authorization or at runtime (`insufficient_scope`).
+Clients may respond to these errors by requesting a new access token with an increased set of scopes via an upgrade flow.
+Clients acting on behalf of a user **SHOULD** attempt the upgrade flow. Clients acting on their own behalf (`client_credentials` clients)
+**MAY** attempt the upgrade flow or abort the request immediately.
+
+The upgrade flow is as follows:
+
+1. **Parse error information** from the authorization server response or `WWW-Authenticate` header
+2. **Determine required scopes** using the following algorithm:
+   - If the `scope` parameter is present in the `WWW-Authenticate` header, use those scopes
+   - If `scope` is not available, use all scopes listed in `scopes_supported` from the Protected Resource Metadata document
+3. **Calculate the union** of current scopes and newly required scopes
+4. **Initiate (re-)authorization** with the scope set
+5. **Retry the original request** with the new authorization no more than a few times and treat this as a permanent authorization failure
+
+Clients **SHOULD** implement retry limits and **SHOULD** cache scope upgrade attempts to avoid repeated failures
+for the same resource and operation combination. The cache should track failed authorization attempts by resource
+and the scopes that were denied, preventing clients from repeatedly requesting the same permissions that were
+previously denied and avoiding unnecessary authorization server requests for known-failing scenarios.
```

Comment:
What is the new default for tool calls after you hit one that needed more auth? Is it the union scopes or the minimum scopes?